supervision/dataset/formats/coco.py

```diff
@@ -0,0 +1,260 @@
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import cv2
+import numpy as np
+
+from supervision.dataset.ultils import approximate_mask_with_polygons
+from supervision.detection.core import Detections
+from supervision.detection.utils import polygon_to_mask
+from supervision.utils.file import read_json_file, save_json_file
+
+
+def _extract_class_names(annotation_data: dict) -> List[str]:
```

Comment:
I noticed a bug in this method. It does not work with the hierarchy of classes. Could you take a look at this example?

```json
[
    {
        "id": 0,
        "name": "fashion-assistant",
        "supercategory": "none"
    },
    {
        "id": 1,
        "name": "baseball cap",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 2,
        "name": "hoodie",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 3,
        "name": "jacket",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 4,
        "name": "pants",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 5,
        "name": "shirt",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 6,
        "name": "shorts",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 7,
        "name": "sneaker",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 8,
        "name": "sunglasses",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 9,
        "name": "sweatshirt",
        "supercategory": "fashion-assistant"
    },
    {
        "id": 10,
        "name": "t-shirt",
        "supercategory": "fashion-assistant"
    }
]
```

Here `fashion-assistant` is a top category - parent to other categories. It should be discarded during loading. So it should be loaded as:

```
[
    'baseball cap',
    'hoodie',
    'jacket',
    'pants',
    'shirt',
    'shorts',
    'sneaker',
    'sunglasses',
    'sweatshirt',
    't-shirt'
]
```

Also, I'm not 100 sure if this implementation guarantee loading class names in the order of class ids.

Reply:
I would even use that example as a case for the unit test.

Reply:
@hardikdava I solved that problem with https://github.com/roboflow/supervision/pull/150/commits/540efc80ce2cfcd37539d35a621575db1f536764